plenum/test/view_change/test_different_last_ordered_before_view_change.py

```diff
@@ -0,0 +1,161 @@
+
+    looper.run(
+        eventually(last_ordered,
+                   txnPoolNodeSet,
+                   slow_instance))
+    last_ordered(txnPoolNodeSet,
+                 txnPoolNodeSet[0].master_replica.instId,
+                 (old_last_ordered[0], old_last_ordered[1] + 1))
+    sdk_send_random_and_check(looper, txnPoolNodeSet, sdk_pool_handle,
+                              sdk_wallet_client, 1)
+    assert all(0 == node.spylog.count(node.request_propagates)
+               for node in txnPoolNodeSet)
+
+
+def test_different_prepares_on_backup_before_view_change(looper, txnPoolNodeSet,
+                                                  sdk_pool_handle, sdk_wallet_client):
+    ''' Send random request and do view change then fast_nodes (2,3 - with
+    primary backup replica) will have prepare or send preprepare on backup
+    replicas and slow_nodes are have not and transaction will ordered on all
+    master replicas. Check last ordered after view change and after another
+    one request.'''
+    sdk_send_random_and_check(looper, txnPoolNodeSet, sdk_pool_handle,
+                              sdk_wallet_client, 1)
+    slow_instance = 1
+    slow_nodes = txnPoolNodeSet[1:3]
+    fast_nodes = [n for n in txnPoolNodeSet if n not in slow_nodes]
+    nodes_stashers = [n.nodeIbStasher for n in slow_nodes]
+    with delay_rules(nodes_stashers, pDelay(delay=sys.maxsize,
+                                            instId=slow_instance)):
+        with delay_rules(nodes_stashers, ppDelay(delay=sys.maxsize,
+                                                 instId=slow_instance)):
+            # send one  request
+            requests = sdk_send_random_requests(looper, sdk_pool_handle,
+                                                sdk_wallet_client, 1)
+            sdk_get_replies(looper, requests)
+            looper.run(
+                eventually(is_prepared,
+                           fast_nodes,
+                           2,
+                           slow_instance))
+
+            # trigger view change on all nodes
+            for node in txnPoolNodeSet:
+                node.view_changer.on_master_degradation()
+
+            # wait for view change done on all nodes
+            ensureElectionsDone(looper, txnPoolNodeSet)
+
+    last_ordered_3pc = fast_nodes[0].replicas[slow_instance].last_ordered_3pc
+    for node in txnPoolNodeSet:
+        assert last_ordered_3pc == node.replicas[slow_instance].last_ordered_3pc
+    sdk_send_random_and_check(looper, txnPoolNodeSet, sdk_pool_handle,
+                              sdk_wallet_client, 1)
+    looper.run(
+        eventually(last_ordered,
+                   txnPoolNodeSet,
+                   slow_instance,
+                   (txnPoolNodeSet[0].viewNo, 1)))
+    assert all(0 == node.spylog.count(node.request_propagates)
+               for node in txnPoolNodeSet)
+
+
+def test_different_last_ordered_on_master_before_view_change(looper, txnPoolNodeSet,
```

Comment:
I would rename the tests to `test_no_propagate_request_on_different_last_ordered_on_master_before_vc`

Reply:
done